### PR TITLE
infrastructure: do npm publish on git tags #38

### PR DESCRIPTION
**Pull request by CAMOBAP**
- #38

### CODE REVIEW

File: .github/workflows/tests.yaml
Comment (e271828-) [inline .github/workflows/tests.yaml:121]:
```suggestion
      - name: npm release
```

File: .github/workflows/tests.yaml
Comment (e271828-) [inline .github/workflows/tests.yaml:115]: should only happen on master

**Comment by e271828-** [comment]
Also need to update docs with a section for maintainers describing release flow. e.g. RELEASES.md

File: .github/workflows/tests.yaml
Comment (CAMOBAP) [reply .github/workflows/tests.yaml]: Done, now we check `startsWith(github.ref, 'refs/tags') && github.event_name == 'push'` and because we handle `push` only for `master` this will filter out other variants

**Review by e271828- — APPROVED** [review]

File: .github/workflows/tests.yaml
Comment (e271828-) [reply .github/workflows/tests.yaml]: Maybe add a comment for this, since it's not an obvious security measure.

**Comment by e271828-** [comment]
One other q:

```
  schedule:
    - cron: '0 0 * * 0' # https://crontab.guru/every-week
```

Why do we need this? It's triggered from branch push.

**Comment by CAMOBAP** [comment]
> One other q:
> 
> ```
>   schedule:
>     - cron: '0 0 * * 0' # https://crontab.guru/every-week
> ```
> 
> Why do we need this? It's triggered from branch push. Is this trying to catch tests breaking on their own?

React native often release a new versions, so I decided to add cron to check compatibility and be acknowledged of possible compatibility issues

File: .github/workflows/tests.yaml
Comment (CAMOBAP) [reply .github/workflows/tests.yaml]: I have re-checked my solution and it doesn't work, so I updated the solution, and now it looks more clear.

So if we will do a tag on the feature branch release will not happen.
BUT also after the merge, it will not happen also, because tags have no "bindings" to branches, they are just aliases to sha.

So I see several options here:
1. We can agree that we will do tags only on the master/main branch (tags on feature branches will be ignored anyway)
2. We can do the release manually (by triggering the release workflow from a button `[workflow_dispatch`](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/) event)
3. We can implement a step as part of the release workflow that will check if the current commit is under a tag

@e271828- let me know how do you think about the options above?

**Review by e271828- — APPROVED** [review]

**Pull request merged**
